core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -515,11 +513,17 @@ private Snapshot createSnapshot(long minGeneration) {
      * while receiving future ones as well
      */
     public Translog.View newView() {
+        long viewGenToClean = -1;
         try (ReleasableLock lock = readLock.acquire()) {
             ensureOpen();
-            View view = new View(lastCommittedTranslogFileGeneration);
-            outstandingViews.add(view);
+            viewGenToClean = deletionPolicy.acquireTranslogGenForView();
```

Comment:
It would be but I'm paranoid about an exception in the View constructor. This way it's clearly safe.